sdk/storage/azfile/directory/client.go

```diff
@@ -93,23 +104,23 @@ func (d *Client) URL() string {
 func (d *Client) NewSubdirectoryClient(subDirectoryName string) *Client {
 	subDirectoryName = url.PathEscape(strings.TrimRight(subDirectoryName, "/"))
 	subDirectoryURL := runtime.JoinPaths(d.URL(), subDirectoryName)
-	return (*Client)(base.NewDirectoryClient(subDirectoryURL, d.generated().Pipeline(), d.sharedKey()))
+	return (*Client)(base.NewDirectoryClient(subDirectoryURL, d.generated().InternalClient(), d.sharedKey(), d.getClientOptions()))
 }
 
 // NewFileClient creates a new file.Client object by concatenating fileName to the end of this Client's URL.
 // The new file.Client uses the same request policy pipeline as the Client.
 func (d *Client) NewFileClient(fileName string) *file.Client {
 	fileName = url.PathEscape(fileName)
 	fileURL := runtime.JoinPaths(d.URL(), fileName)
-	return (*file.Client)(base.NewFileClient(fileURL, d.generated().Pipeline(), d.sharedKey()))
+	return (*file.Client)(base.NewFileClient(fileURL, d.generated().InternalClient(), d.sharedKey(), d.getClientOptions()))
```

Comment:
The file client will end up with the wrong name using in distributed tracing (see line 52 above).

Reply:
Can we use the name "azfile.Client" instead of "directory.Client", "file.Client", etc. for all packages.
Creating new azcore client will create new pipelines. So, if file client is created from directory client, both will have different pipelines. Correct me if I am wrong here. 
Ideal scenario will be the child and parent clients sharing the same pipeline like it is currently with the old code generator.

Reply:
@jhendrixMSFT, can you confirm if the above change looks good.

Reply:
IMO the client names need to be unique so that when looking at a trace it's clear what the spans apply to (especially if any clients have the same method names).  Including @lmolkova for comments.

As I recall, sharing the pipeline was to simplify creating the "child" client with respect to credentials.  Other than that there's no need to share them right?

Reply:
Per today's discussion, we can create a new `*azcore.Client` with the correct client name.

I've opened https://github.com/Azure/azure-sdk-for-go/issues/20973 to track optimizing this so we can continue to share pipelines across child clients.

Reply:
@jhendrixMSFT, I have made the changes. Can you please review it?

Reply:
Looks great!